### PR TITLE
refine IANA table, add definition of DNS operator policy and network …

### LINKED ISSUE #1: Rationale and References for assigning a suberror code
Add some text to clarify the reasoning for the listed suberrors. 

A proposal is to echo the following in the text:  say basically that we can accept registering suberrors:
  *  for which a clear definition is provided and
  *  which provide more characterization of an error vs a “parent” error (8914).
 
Most of the current proposed entries in the draft have their counterparts defined in existing RFCs (RFC4949, RFC5901). These citations should be added to the draft.

### LINKED ISSUE #2: Add "Policy imposed by the operator of the DNS server" suberror
as per the discussion with Neil

### PR DESCRIPTION
**Pull request by danwing**
…operator policy, remove Abuse

### CODE REVIEW

File: draft-ietf-dnsop-structured-dns-error.md [hunk 1]
```diff
@@ -417,6 +416,53 @@ via {{?I-D.reddy-add-resolver-info}} and the client includes the EDE OPT
...
+
+## Network Operator Policy {#policy-network}
+
+  * Number: 5
+
+  * Meaning: Network Operator Policy
+
+  * Applicability: Blocked, Forged
+
+  * Reference: the operator providing IP connectivity filtered the response
```
Comment (neilcook) [inline draft-ietf-dnsop-structured-dns-error.md]:
```suggestion
  * Reference: Filtered according to policy determined by the operator of the local network 
```

File: draft-ietf-dnsop-structured-dns-error.md [hunk 2]
```diff
@@ -417,6 +416,53 @@ via {{?I-D.reddy-add-resolver-info}} and the client includes the EDE OPT
...
+
+## DNS Operator Policy {#policy-dns}
+
+  * Number: 6
+
+  * Meaning: DNS Operator Policy
+
+  * Applicability: Blocked, Forged
+
+  * Reference: the operator of the DNS server filtered the response
```
Comment (neilcook) [inline draft-ietf-dnsop-structured-dns-error.md]:
```suggestion
  * Reference:  Filtered according to policy determined by the operator of the DNS server
```

File: draft-ietf-dnsop-structured-dns-error.md [hunk 3]
```diff
@@ -417,6 +416,53 @@ via {{?I-D.reddy-add-resolver-info}} and the client includes the EDE OPT
 pseudo-RR in the query. This allows the server to differentiate
 EDE-aware clients from EDE-unaware clients and respond appropriately.
 
+# New Sub-Error Codes Definition
+
+Subsequent documents defining new Sub-Error codes MUST be accepted by
```
Comment (boucadair) [inline draft-ietf-dnsop-structured-dns-error.md]: This text conflicts with the policy set for the registry (expert review).  I think the intent here is not about the document that requests the new code but the reference that is provided to characterize the error. I agree that it is preferable  to 
 have an IETF ref, but MUST is too strong here.

File: draft-ietf-dnsop-structured-dns-error.md [hunk 4]
```diff
@@ -417,6 +416,53 @@ via {{?I-D.reddy-add-resolver-info}} and the client includes the EDE OPT
...
+IETF, MUST also provide a clear definition of the error code (in their
+Reference text), and MUST also provide more characterization than the
```
Comment (boucadair) [inline draft-ietf-dnsop-structured-dns-error.md]:
```suggestion
Meaning or Reference clauses), and MUST also provide more characterization than the
```

File: draft-ietf-dnsop-structured-dns-error.md [hunk 5]
```diff
@@ -417,6 +416,53 @@ via {{?I-D.reddy-add-resolver-info}} and the client includes the EDE OPT
...
+
+This document defines new IANA-registered Sub-Error codes, below.
+
+## Reserved {#policy-reserved}
+
+  * Number: 0
+
+  * Meaning: Reserved
+
+  * Applicability: Blocked, Forged, Censored, Filtered
```
Comment (boucadair) [inline draft-ietf-dnsop-structured-dns-error.md]:
```suggestion
  * Applicability: N/A. 
```

This code must not be used at all.

File: draft-ietf-dnsop-structured-dns-error.md [hunk 6]
```diff
@@ -417,6 +416,53 @@ via {{?I-D.reddy-add-resolver-info}} and the client includes the EDE OPT
...
+
+  * Reference: this sub-error code value MUST NOT be sent. If received, it has no meaning.
```
Comment (boucadair) [inline draft-ietf-dnsop-structured-dns-error.md]: This text should be under "meaning".

File: draft-ietf-dnsop-structured-dns-error.md [hunk 7]
```diff
@@ -539,25 +585,26 @@ following fields:
...
 * Number: Wire format suberror code (range 0-255)
 
-* Meaning: A short description of the error
+* Meaning: A short description of the sub-error
 
-* Applicable EDO Codes: List the Extended DNS Error codes to which the suberror code applies.
+* Applicability: Indicates which RFC8914 error codes apply to this sub-error code
 
-* Reference: A pointer to the specification text
+* Reference: A pointer to IETF-approved specification
```
Comment (boucadair) [inline draft-ietf-dnsop-structured-dns-error.md]:
```suggestion
* Reference: A pointer to the specification that registered the code and/or an authoritative specification that describes the meaning of this code.
```

File: draft-ietf-dnsop-structured-dns-error.md [hunk 8]
```diff
@@ -539,25 +585,26 @@ following fields:
...
-| 0 | Reserved | N/A| This-Document | IETF |
-| 1 | Malware | "Censored", "Filtered", "Blocked", and "Forged" |This-Document | IETF |
-| 2 | Phishing | "Censored", "Filtered", "Blocked", and "Forged" |This-Document | IETF |
-| 3 | Spam  | "Censored", "Filtered", "Blocked", and "Forged" |This-Document | IETF |
-| 4 | Spyware | "Censored", "Filtered", "Blocked", and "Forged" |This-Document | IETF |
-| 5 | Adware | "Censored", "Filtered", "Blocked", and "Forged" |This-Document | IETF |
-| 6 | Network policy imposed by the operator of the network | "Blocked", and "Forged" |This-Document | IETF |
+| Number | Meaning | RFC8914 error code applicability | Reference |  Change Controller |
+|:------:|:--------|:---------------------------------|:----------|:------------------:|
+| 0 | Reserved| Blocked, Forged, Censored, Filtered | {{policy-reserved}} | IETF |
```
Comment (boucadair) [inline draft-ietf-dnsop-structured-dns-error.md]: I would also list this document under Reference to keep track of documents that registered a code.

File: draft-ietf-dnsop-structured-dns-error.md (hunk 4 above)
Comment (danwing) [reply draft-ietf-dnsop-structured-dns-error.md]: I have removed my text which was intended as guidance to the expert reviewer.

File: draft-ietf-dnsop-structured-dns-error.md (hunk 5 above)
Comment (danwing) [reply draft-ietf-dnsop-structured-dns-error.md]: Ok, updating doc accordingly.

**Pull request merged**